libs/cart/src/reducers/loading/cart-loading.type.ts

```diff
@@ -2,6 +2,29 @@ import { DaffLoadingState } from '@daffodil/core/state';
 
 import { DaffCartOperationType } from '../cart-operation-type.enum';
 
-export type DaffCartLoading = {
-  [K in DaffCartOperationType]: DaffLoadingState
+export interface DaffCartLoading {
+	[DaffCartOperationType.Cart]: DaffLoadingState;
+	[DaffCartOperationType.Item]: DaffCartItemLoadingState;
+	[DaffCartOperationType.BillingAddress]: DaffLoadingState;
+	[DaffCartOperationType.ShippingAddress]: DaffLoadingState;
+	[DaffCartOperationType.Payment]: DaffLoadingState;
+	[DaffCartOperationType.PaymentMethods]: DaffLoadingState;
+	[DaffCartOperationType.ShippingInformation]: DaffLoadingState;
+	[DaffCartOperationType.ShippingMethods]: DaffLoadingState;
+	[DaffCartOperationType.Coupon]: DaffLoadingState;
 }
+
+export enum DaffCartItemLoadingState {
+  Resolving = 'Resolving',
+	Mutating = 'Mutating',
+	Adding = 'Adding',
+  Complete = 'Complete',
+}
+
+export const initializeLoadingSetter = (loadingSpace: DaffCartOperationType) =>
```

Comment:
I had to move this here, because if it isn't in the same file as `DaffCartLoading`, the following error is thrown: https://github.com/microsoft/TypeScript/issues/5711